docs/source/configuration/entity-caching.mdx

```diff
@@ -129,9 +128,162 @@ This entry contains an object with the `all` field to affect all subgraph reques
+
+</Note>
+
+
+#### Invalidation HTTP endpoint
+
+The invalidation endpoint exposed by the router can receive multiple invalidation requests at once and will process them in sequence. For authorization, you can provide a shared key in the request. For example, with the previous configuration you can send the following request:
+
+```
+POST http://127.0.0.1:3000/invalidation
+Authorization: agm3ipv7egb78dmxzv0gr5q0t5l6qs37
+Content-Length:96
+Content-Type:application/json
+Accept: application/json
+
+[{
+    "kind": "type",
+    "subgraph": "invalidation-subgraph-type-accounts",
+    "type": "Query"
+}]
+```
+
+We would expect the following response:
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "count": 300
+}
+```
+
+The `count` field indicates the number of keys that were removed from Redis.
 
+#### Invalidation through subgraph response extensions
+
+A subgraph can return an `invalidation` array with invalidation requests in its response's `extensions` field.
+
+```json
+{
+  "data": { "invalidateProductReview": 1 },
+  "extensions": {
+      "invalidation": [{
+          "kind": "entity",
+          "subgraph": "invalidation-entity-key-reviews",
+          "type": "Product",
+          "key": {
+              "upc": "1"
+          }
+      }]
+  }
+}
 ```
 
+#### Observability
+
+Invalidation requests are instrumented with the following metrics:
+- `apollo.router.operations.entity.invalidation.event` - counter triggered when a batch of invalidation requests is received. It has a label `origin` that can be either `endpoint` or `extensions`.
+- `apollo.router.operations.entity.invalidation.entry` - counter measuring how many entries are removed per `DEL` call. It has a label `origin` that can be either `endpoint` or `extensions`, and a label `subgraph.name` with the name of the receiving subgraph.
+- `apollo.router.cache.invalidation.keys` - histogram measuring the number of keys that were removed from Redis per invalidation request.
+- `apollo.router.cache.invalidation.duration` - histogram measuring the time spent handling one invalidation request.
+
+Invalidation requests are also reported under the following spans:
+- `cache.invalidation.batch` - span covering the processing of a list of invalidation requests. It has a label `origin` that can be either `endpoint` or `extensions`.
+- `cache.invalidation.request` - span covering the processing of a single invalidation request.
+
+### Failure cases
+
+Entity caching will greatly reduce traffic to subgraphs, resulting in less infrastructure to serve more client traffic. In the rare case where Redis is unavailable, the router should be configured with [rate limiting for subgraph requests](/router/configuration/traffic-shaping/#rate-limiting-1) to avoid overwhelming the subgraphs. It can also be paired with [subgraph query deduplication](/router/configuration/traffic-shaping/#query-deduplication) to further reduce traffic.
```

Comment:
```suggestion
Entity caching will greatly reduce traffic to subgraphs. Should there be an availability issue with a Redis cache, this could cause traffic to subgraphs to increase to a level where infrastructure becomes overwhelmed. To avoid such issues, the router should be configured with [rate limiting for subgraph requests](/router/configuration/traffic-shaping/#rate-limiting-1) to avoid overwhelming the subgraphs. It could also be paired with [subgraph query deduplication](/router/configuration/traffic-shaping/#query-deduplication) to further reduce traffic.
```